modules/ROOT/pages/migration-from-5x.adoc

```diff
@@ -3,88 +3,585 @@
+
+=== `disabled` functions renamed in all {productname} APIs
+
+Previously, multiple {productname} APIs and UI classes used `disable` or `disabled` in their function or property names.
+
+In {productname} 6, these function names have been changed and configuration changes are required.
+
+. `isDisabled()` functions are now `isEnabled()` functions.
++
+Configurations that called `isDisabled()` must now call `isEnabled()`. Also, the returned or set value must be interpreted in terms of the new function name. For example, if `isDisabled(false)` was previously called, the equivalent new call is `isEnabled(true)`.
++
+. The `enable()` and `disable()` functions are now the single `setEnabled(<state>)` function.
++
+Rather than calling the function to set the desired state, call `setEnabled()` with the `true` or `false` value to set the desired state.
++
+. The `disable(<name>)` and `enable(<name>)` functions are now the single `setEnabled(<name>, <state>)` function in the Dialog APIs.
++
+As well, the `.disabled` property is now the `.enabled` property in all {productname} UI classes.
+
+=== `editor.SetContent`
+
+The `editor.SetContent` event’s `content` property has been deprecated.
+
+When the `editor.SetContent` event fires, any data in `content` has already been added to the {productname} editor instance and cannot be altered, making `content` essentially moot as an `editor.SetContent` property.
+
+To alter such data before it is added to the {productname} editor, use the `content` property available in the `BeforeSetContent` event.
+
+NOTE: Calling the `editor.SetContent` `content` property now returns an empty string.
+
+=== `fire` function renamed in all {productname} APIs
+
+The `fire()` function in
+
+* `tinymce.Editor`;
+* `tinymce.dom.EventUtils`;
+* `tinymce.dom.DOMUtils`;
+* `tinymce.util.Observable`; and
+* `tinymce.util.EventDispatcher`
+
+has been renamed to `dispatch()`.
+
+`fire` has been aliased to `dispatch` but has also been marked as _deprecated_.
+
+=== `new` keyword no longer used to instantiate plugins
+
+In {productname} 6.0, plugins (and other addons) are no longer initialised with the `new` keyword.
+
+As a consequence, the plugin initalizer value passed to `PluginManager.add` must be a function. It cannot be a class. (The plugin initializer value is the second argument passed to `PluginManager.add()`.)
+
+This change makes it safe to use both older functions and newer arrow functions when instantiating add-ons.
+
+=== PluginManager
+
+Previously, when loading a requested resource, the `PluginManager` passed a `success` or `failure` callback to report the status of the request. In {productname} 6, `PluginManager` reports this status by returning a `Promise`, with a value of either `resolve` or `reject`.
+
+=== redial
```

Comment:
```suggestion
=== Dialog `redial` API
```
"redial" is very much a TinyMCE concept, and specific to dialogs, so I figured this is more accurate